tests/cases/fourslash/goToDefinition_filteringMappedType.ts

```diff
@@ -0,0 +1,7 @@
+///<reference path="fourslash.ts"/>
+
+////const obj = { /*def*/a: 1, b: 2 };
+////const filtered: { [P in keyof typeof obj as P extends 'b' ? never : P]: 0; } = { a: 0 };
+////filtered.[|/*ref*/a|];
+
+verify.goToDefinition("ref", "def");
```

Comment:
I try to add a test for symbol renaming too but I can't figure out how to write one properly. Any tips would be highly appreciated.

So far I have this:
```ts
/// <reference path='fourslash.ts' />

////const obj = { [|[|{| "contextRangeIndex": 0 |}a|]: string,|] b: 2 };
////const filtered: { [P in keyof typeof obj as P extends 'b' ? never : P]: 0; } = { a: 0 };
////filtered.[|a|]();

const [r0Def, r0, r1] = test.ranges();
verify.renameLocations(r0, [r0, r1]);
verify.renameLocations(r1, [r0, r1]);
```

Reply:
Note that it works OK in the IDE as I've confirmed that by using the `built/local/tsserver.js`.

Reply:
It seems that I'm using this whole `contextRangeIndex` incorrectly. I've tried without it too but no luck so far.

Reply:
Generally if find-all-refs works then goto-def and rename work too. Can you write a similar test that uses `verify.baselineFindAllReferences` from at least the last line, but preferably the first and second. I think usage of that function is simpler and you can just prefix the locations with `/*1*/`, `/*2*/`, etc.

Reply:
Thanks for the tip! I've pushed out such a test and it looks correct to me. Note that this was the first time I've used this API so I'm not exactly sure how to interpret all of the information in the generated baseline but the `references` property for all 3 markers looks alright to me.